### DISCUSSION TITLE
How can I use the command in this situation when needs to create one of two objects

### DISCUSSION THREAD

**Discussion by itprodavets** [jasontaylordev/CleanArchitecture#1004] [Show and tell]
I have a situation where two objects are created separately, but when one object should be created with the second, what should I do?

```C#

ACommand : IRequest 
{
    public string Name { get; }
    public string Description { get; }
}

BCommand : IRequest
{
    public string Name { get; }
    public string Description { get; }
}
```

1. Create a new command with list the properties of the first and second object.

```C#

ABCommand : IRequest 
{
    public string AName { get; }
    public string ADescription { get; }
    public string BName { get; }
    public string BDescription { get; }
}
```

2. Create a team where there will be the properties of the teams of the first object and the second object

```C#

CCommand : IRequest 
{
    public ACommand A { get; }
    public BCommand B { get; }
}
```

**Comment by MedBouc** [comment]
Can you please reformulate your question.
Do you want to create both objects or only one of the two?
I guess you want to automatically create a second object when a first one is created.
What prevents you from creating them one after the other?
During the validation process of the first one, you can check and ask for the creation of the second one if it does not exist.

**Comment by itprodavets** [comment]
Dear,

``` C#
public class CreateSensorCommand : IRequest<string>
{
    public CreateSensorCommand (
        string name,
        string description,
        IEnumerable<CreateRegisterCommand> registers
    )
    {
        Name = name;
        Description = description;
        Registers = registers;
    }

    public string Name { get; }
    public string Description { get; }
    public IEnumerable<CreateRegisterCommand> Registers { get; }
}
```

Here is an example where I create a new sensor in the system and add many registers . The register itself can be created without a sensor, but vice versa is not possible.

How correct is it to use a command within a command?
